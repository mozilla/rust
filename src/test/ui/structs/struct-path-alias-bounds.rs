// issue #36286

struct S<T: Clone> { a: T }

struct NoClone;
type A = S<NoClone>; //~ ERROR the trait bound `NoClone: Clone` is not satisfied

fn main() {
    let s = A { a: NoClone };
}
