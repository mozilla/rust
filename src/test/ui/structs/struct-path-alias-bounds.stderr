error[E0277]: the trait bound `NoClone: Clone` is not satisfied
  --> $DIR/struct-path-alias-bounds.rs:6:10
   |
LL | struct S<T: Clone> { a: T }
   |             ----- required by this bound in `S`
...
LL | type A = S<NoClone>;
   |          ^^^^^^^^^^ the trait `Clone` is not implemented for `NoClone`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
